### ISSUE TITLE
Crash when locked and monitor turned off

### ISSUE THREAD

**Issue by pmkap** [riverwm/river#809]
I locked the session with swaylock. I left. The monitor turns off automatically after some time. When I came back after some hours, river had crashed.

~~This does not seem to happen with waylock, only swaylock.~~
Correction, it happens both with swaylock and waylock.

* River: build from master
* Wayland, Waylock, Swaylock: latest from Arch Linux repos

```
Core was generated by `river'.
Program terminated with signal SIGABRT, Aborted.
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
Downloading source file /usr/src/debug/glibc/glibc/nptl/pthread_kill.c
44	    return INTERNAL_SYSCALL_ERROR_P (ret) ? INTERNAL_SYSCALL_ERRNO (ret) : 0;                                                                                                          
[Current thread is 1 (Thread 0x7fb353e81980 (LWP 316735))]
(gdb) bt
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007fb3548e0953 in __pthread_kill_internal (signo=6, threadid=<optimized out>) at pthread_kill.c:78
#2  0x00007fb354891ea8 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007fb35487b53d in __GI_abort () at abort.c:79
#4  0x0000000000286416 in os.abort () at /usr/lib/zig/std/os.zig:560
#5  0x0000000000286cf2 in debug.panicImpl (trace=<optimized out>, first_trace_addr=..., msg=...) at /usr/lib/zig/std/debug.zig:384
#6  0x0000000000234e77 in builtin.default_panic (msg=..., error_return_trace=0x0, ret_addr=...) at /usr/lib/zig/std/builtin.zig:839
#7  0x000000000025416b in LockSurface.handleMap () at LockSurface.zig:89
#8  wayland_server_core.Listener(void).setNotify__struct_14560.wrapper (listener=<optimized out>)
    at /home/peter/.cache/paru/clone/river-git/src/river/zig-cache/zig-wayland/wayland_server_core.zig:553
#9  0x00007fb354c41f6e in wl_signal_emit_mutable (signal=<optimized out>, data=0x0) at ../wayland-1.21.0/src/wayland-server.c:2179
#10 0x00007fb354ab3e9d in surface_commit_state (surface=0xe3a670, next=0xe3a7f8) at ../wlroots-0.16.2/types/wlr_compositor.c:488
#11 0x00007fb3546fc4f6 in ffi_call_unix64 () at ../src/x86/unix64.S:104
#12 0x00007fb3546f8f5e in ffi_call_int (cif=cif@entry=0x7ffd411c6000, fn=<optimized out>, rvalue=<optimized out>, avalue=<optimized out>, closure=closure@entry=0x0) at ../src/x86/ffi64.c:673
#13 0x00007fb3546fbb73 in ffi_call (cif=cif@entry=0x7ffd411c6000, fn=<optimized out>, rvalue=rvalue@entry=0x0, avalue=avalue@entry=0x7ffd411c60d0) at ../src/x86/ffi64.c:710
#14 0x00007fb354c40ada in wl_closure_invoke (closure=closure@entry=0xce7c90, target=<optimized out>, target@entry=0xe49560, opcode=opcode@entry=6, data=<optimized out>, 
    data@entry=0xe326b0, flags=2) at ../wayland-1.21.0/src/connection.c:1025
#15 0x00007fb354c4501c in wl_client_connection_data (fd=<optimized out>, mask=<optimized out>, data=<optimized out>) at ../wayland-1.21.0/src/wayland-server.c:437
#16 0x00007fb354c439e2 in wl_event_loop_dispatch (loop=0x34f400, timeout=timeout@entry=-1) at ../wayland-1.21.0/src/event-loop.c:1027
#17 0x00007fb354c44197 in wl_display_run (display=0x34f310) at ../wayland-1.21.0/src/wayland-server.c:1431
#18 0x0000000000234995 in main.main () at main.zig:132
#19 start.callMain () at /usr/lib/zig/std/start.zig:614
#20 start.initEventLoopAndCallMain () at /usr/lib/zig/std/start.zig:548
#21 start.callMainWithArgs () at /usr/lib/zig/std/start.zig:498
#22 main (c_argc=<optimized out>, c_argv=<optimized out>, c_envp=<optimized out>) at /usr/lib/zig/std/start.zig:513
```
Cheers!

**Comment by MaxVerevkin** [comment]
Probably related: river crashes when a monitor is hot-plugged while session is locked:

```
(gdb) bt full
#0  0x00007f64636508ec in  () at /lib64/libc.so.6
#1  0x00007f6463601ea8 in raise () at /lib64/libc.so.6
#2  0x00007f64635eb53d in abort () at /lib64/libc.so.6
#3  0x00000000002a6fc9 in os.abort () at /usr/lib/zig/std/os.zig:560
#4  0x0000000000262968 in debug.panicImpl (trace=0x0, first_trace_addr=..., msg=...) at /usr/lib/zig/std/debug.zig:384
#5  0x000000000024eb0a in builtin.default_panic (msg=..., error_return_trace=0x0, ret_addr=...) at /usr/lib/zig/std/builtin.zig:839
        first_trace_addr = 2428567
#6  0x0000000000250e97 in debug.assert (ok=false) at /usr/lib/zig/std/debug.zig:278
#7  0x00000000002793c9 in LockSurface.handleMap () at LockSurface.zig:105
        lock_surface = 0x1989540
        output = 0x1844800
        listener = 0x1989568
#8  wayland_server_core.Listener(void).setNotify__struct_14575.wrapper (listener=0x1989568) at /home/maxv/river/zig-cache/zig-wayland/wayland_server_core.zig:553
#9  0x00007f64639b1f6e in wl_signal_emit_mutable () at /lib64/libwayland-server.so.0
#10 0x00007f6463823e9d in  () at /lib64/libwlroots.so.11
#11 0x00007f646346f4f6 in  () at /usr/lib/libffi.so.8
#12 0x00007f646346bf5e in  () at /usr/lib/libffi.so.8
#13 0x00007f646346eb73 in ffi_call () at /usr/lib/libffi.so.8
#14 0x00007f64639b0ada in  () at /lib64/libwayland-server.so.0
#15 0x00007f64639b501c in  () at /lib64/libwayland-server.so.0
#16 0x00007f64639b39e2 in wl_event_loop_dispatch () at /lib64/libwayland-server.so.0
#17 0x00007f64639b4197 in wl_display_run () at /lib64/libwayland-server.so.0
#18 0x000000000026072c in main.main () at main.zig:132
        startup_command = {ptr = 0xc882e0 "English (Dvorak, right-handed)", len = 29}
        result = {args = {ptr = 0x7fffb0ec91c0, len = 0}, flags = {h = false, version = false, c = {ptr = 0x0, len = 0}, log-level = {ptr = 0x7fffb0ecab01 "debug", len = 5}}}
        child_pgid = {data = 845, some = 1 '\001'}
```

```
debug(transaction): commiting transaction
made it here ig
debug(wlroots): [types/wlr_compositor.c:673] New wlr_surface 0x1a247b0 (res 0x199fac0)
debug(session_lock): new ext_session_lock_surface_v1 created
debug(wlroots): [types/wlr_compositor.c:673] New wlr_surface 0x19db360 (res 0x18cbd40)
info(session_lock): session locked
debug(wlroots): [backend/session/session.c:181] udev event for card0 (change)
debug(wlroots): [backend/session/session.c:210] DRM device card0 changed
debug(wlroots): [backend/drm/backend.c:155] Received hotplug event for /dev/dri/card0
info(wlroots): [backend/drm/drm.c:1335] Scanning DRM connector 85 on /dev/dri/card0
info(wlroots): [backend/drm/drm.c:1414] 'HDMI-A-1' connected
debug(wlroots): [backend/drm/drm.c:1220] Current CRTC: -1
info(wlroots): [backend/drm/drm.c:1230] Physical size: 600x340
info(wlroots): [backend/drm/drm.c:1287] Detected modes:
info(wlroots): [backend/drm/drm.c:1315]   1920x1080 @ 60.000 Hz (preferred)
info(wlroots): [backend/drm/drm.c:1315]   1920x1080 @ 74.973 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1920x1080 @ 60.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1920x1080 @ 59.940 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1920x1080 @ 50.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1680x1050 @ 59.883 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1600x900 @ 60.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1280x1024 @ 75.025 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1280x1024 @ 60.020 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1440x900 @ 59.901 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1280x800 @ 59.910 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1152x864 @ 75.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1280x720 @ 60.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1280x720 @ 60.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1280x720 @ 59.940 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1280x720 @ 50.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1280x720 @ 50.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1024x768 @ 75.029 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1024x768 @ 70.069 Hz 
info(wlroots): [backend/drm/drm.c:1315]   1024x768 @ 60.004 Hz 
info(wlroots): [backend/drm/drm.c:1315]   832x624 @ 74.551 Hz 
info(wlroots): [backend/drm/drm.c:1315]   800x600 @ 75.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   800x600 @ 72.188 Hz 
info(wlroots): [backend/drm/drm.c:1315]   800x600 @ 60.317 Hz 
info(wlroots): [backend/drm/drm.c:1315]   800x600 @ 56.250 Hz 
info(wlroots): [backend/drm/drm.c:1315]   720x576 @ 50.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   720x576 @ 50.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   720x480 @ 60.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   720x480 @ 60.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   720x480 @ 59.940 Hz 
info(wlroots): [backend/drm/drm.c:1315]   720x480 @ 59.940 Hz 
info(wlroots): [backend/drm/drm.c:1315]   640x480 @ 75.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   640x480 @ 72.809 Hz 
info(wlroots): [backend/drm/drm.c:1315]   640x480 @ 66.667 Hz 
info(wlroots): [backend/drm/drm.c:1315]   640x480 @ 60.000 Hz 
info(wlroots): [backend/drm/drm.c:1315]   640x480 @ 59.940 Hz 
info(wlroots): [backend/drm/drm.c:1315]   720x400 @ 70.082 Hz 
debug(wlroots): [backend/drm/drm.c:1011] Reallocating CRTCs
debug(wlroots): [backend/drm/drm.c:1022] State before reallocation:
debug(wlroots): [backend/drm/drm.c:1036]   'eDP-1' crtc=0 status=1 want_crtc=1
debug(wlroots): [backend/drm/drm.c:1036]   'HDMI-A-1' crtc=-1 status=1 want_crtc=0
debug(wlroots): [backend/drm/drm.c:1078] State after reallocation:
debug(wlroots): [backend/drm/drm.c:1085]   'eDP-1' crtc=0 status=1
debug(wlroots): [backend/drm/drm.c:1085]   'HDMI-A-1' crtc=-1 status=1
info(wlroots): [backend/drm/drm.c:1447] connector HDMI-A-1: Requesting modeset
debug(output_manager): new output HDMI-A-1
debug(wlroots): [types/output/render.c:229] Attaching empty buffer to output for modeset
debug(wlroots): [backend/drm/drm.c:1011] Reallocating CRTCs
debug(wlroots): [backend/drm/drm.c:1022] State before reallocation:
debug(wlroots): [backend/drm/drm.c:1036]   'eDP-1' crtc=0 status=1 want_crtc=1
debug(wlroots): [backend/drm/drm.c:1036]   'HDMI-A-1' crtc=-1 status=1 want_crtc=1
debug(wlroots): [backend/drm/drm.c:1078] State after reallocation:
debug(wlroots): [backend/drm/drm.c:1085]   'eDP-1' crtc=0 status=1
debug(wlroots): [backend/drm/drm.c:1085]   'HDMI-A-1' crtc=3 status=1
debug(wlroots): [types/output/render.c:75] Choosing primary buffer format XR24 (0x34325258) for output 'HDMI-A-1'
debug(wlroots): [render/swapchain.c:105] Allocating new swapchain buffer
debug(wlroots): [render/allocator/gbm.c:143] Allocated 1920x1080 GBM buffer with format XR24 (0x34325258), modifier GFX9,GFX9_64K_S_X,DCC,DCC_RETILE,DCC_INDEPENDENT_64B,DCC_MAX_COMPRESSED_BLOCK=64B,DCC_CONSTANT_ENCODE,PIPE_XOR_BITS=2,BANK_XOR_BITS=0,RB=1,PIPE_2 (0x0200000440517901)
debug(wlroots): [render/gles2/renderer.c:148] Created GL FBO for buffer 1920x1080
info(wlroots): [backend/drm/drm.c:586] connector HDMI-A-1: Modesetting with 1920x1080 @ 60.000 Hz
debug(layout): starting layout demand 'my-layout' on output 'eDP-1'
debug(wlroots): [render/swapchain.c:105] Allocating new swapchain buffer
debug(wlroots): [render/allocator/gbm.c:143] Allocated 1920x1080 GBM buffer with format XR24 (0x34325258), modifier GFX9,GFX9_64K_S_X,DCC,DCC_RETILE,DCC_INDEPENDENT_64B,DCC_MAX_COMPRESSED_BLOCK=64B,DCC_CONSTANT_ENCODE,PIPE_XOR_BITS=2,BANK_XOR_BITS=0,RB=1,PIPE_2 (0x0200000440517901)
debug(wlroots): [render/gles2/renderer.c:148] Created GL FBO for buffer 1920x1080
debug(layout): layout 'my-layout' on output 'eDP-1' pushed view dimensions: 0 0 1601 875
debug(layout): layout 'my-layout' on output 'eDP-1' commited
debug(transaction): commiting transaction
debug(wlroots): [types/wlr_compositor.c:673] New wlr_surface 0x19da100 (res 0x1626570)
debug(wlroots): [types/wlr_layer_shell_v1.c:460] new layer_surface 0x15dac10 (res 0x1625130)
debug(wlroots): [types/wlr_compositor.c:673] New wlr_surface 0x18175c0 (res 0x1624b90)
debug(session_lock): new ext_session_lock_surface_v1 created
debug(wlroots): [types/wlr_compositor.c:673] New wlr_surface 0x1709240 (res 0x19a4aa0)
info(output_manager): applying output configuration
info(wlroots): [backend/drm/drm.c:590] connector eDP-1: Turning off
debug(layout): destroying layout 'my-layout' on output 'eDP-1'
debug(layer_shell): layer surface 'i3bar-river' unmapped
debug(transaction): commiting transaction
debug(layer_shell): layer surface 'i3bar-river' destroyed
debug(transaction): commiting transaction
debug(layout): bind layout 'my-layout' on output 'HDMI-A-1'
debug(layout): starting layout demand 'my-layout' on output 'HDMI-A-1'
debug(server): new layer surface: namespace i3bar-river, layer top, anchor 1101, size 0,20, margin 0,0,5,0, exclusive_zone 25
debug(layout): layout 'my-layout' on output 'HDMI-A-1' pushed view dimensions: 0 0 1920 1080
debug(layout): layout 'my-layout' on output 'HDMI-A-1' commited
debug(transaction): started transaction with 1 pending configure(s)
thread 794 panic: reached unreachable code
/usr/lib/zig/std/debug.zig:278:14: 0x250e96 in assert (river)
    if (!ok) unreachable; // assertion failure
             ^
/home/maxv/river/river/LockSurface.zig:105:38: 0x2793c8 in wrapper (river)
    assert(output.normal_content.node.enabled);
                                     ^
???:?:?: 0x7f64639b1f6d in ??? (???)
```

**Comment by ifreund** [comment]
Should be fixed by 3865a7be7c40d0072523d1f82da78f0ceef617c9
